epsagon/aws_v2_wrapper.go

```diff
@@ -0,0 +1,107 @@
+package epsagon
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/epsagon/epsagon-go/epsagon/aws_sdk_v2_factories"
+	"github.com/epsagon/epsagon-go/protocol"
+	"github.com/epsagon/epsagon-go/tracer"
+	"log"
+	"reflect"
+	"time"
+)
+
+// WrapAwsV2Service wrap aws service with epsgon
+// svc := epsagon.WrapAwsV2Service(dynamodb.New(cfg)).(*dynamodb.Client)
+func WrapAwsV2Service(svcClient interface{}) interface{} {
+	awsClient := reflect.ValueOf(svcClient).Elem().FieldByName("Client").Interface().(*aws.Client)
+	awsClient.Handlers.Complete.PushFrontNamed(
+		aws.NamedHandler{
+			Name: "epsagon-aws-sdk-v2",
+			Fn:   completeEventData,
+		},
+	)
+	return svcClient
+}
+
+func getTimeStampFromRequest(r *aws.Request) float64 {
+	return float64(r.Time.UTC().UnixNano()) / float64(time.Millisecond) / float64(time.Nanosecond) / 1000.0
+}
+
+func completeEventData(r *aws.Request) {
+	defer GeneralEpsagonRecover("aws-sdk-go wrapper", "")
+	log.Printf("EREZ DEBUG REMOVE ME tracer.GetGlobalTracerConfig().Debug = %v", tracer.GetGlobalTracerConfig().Debug)
+	if tracer.GetGlobalTracerConfig().Debug {
+		log.Printf("EPSAGON DEBUG OnComplete request response: %+v\n", r.HTTPResponse)
+		log.Printf("EPSAGON DEBUG OnComplete request Operation: %+v\n", r.Operation)
+		log.Printf("EPSAGON DEBUG OnComplete request Endpoint: %+v\n", r.Endpoint)
+		log.Printf("EPSAGON DEBUG OnComplete request Params: %+v\n", r.Params)
+		log.Printf("EPSAGON DEBUG OnComplete request Data: %+v\n", r.Data)
+	}
+
+	endTime := tracer.GetTimestamp()
+	event := protocol.Event{
+		Id:        r.RequestID,
+		StartTime: getTimeStampFromRequest(r),
+		Origin:    "aws-sdk",
+		Resource:  extractResourceInformation(r),
+	}
+	event.Duration = endTime - event.StartTime
+	tracer.AddEvent(&event)
+}
+
+type factory func(*aws.Request, *protocol.Resource, bool)
+
+var awsResourceEventFactories = map[string]factory{
```

Comment:
I think this might be failing the test, as the other factories were split to a different pr